presto-main/src/test/java/com/facebook/presto/operator/TestHashSemiJoinOperator.java

```diff
@@ -130,6 +142,36 @@ public void testSemiJoin(boolean hashEnabled)
         OperatorAssertion.assertOperatorEquals(joinOperatorFactory, driverContext, probeInput, expected, hashEnabled, ImmutableList.of(probeTypes.size()));
     }
 
+    @Test(dataProvider = "dataType")
+    public void testSemiJoinMemoryReservationYield(Type type)
+    {
+        // We only need the first column so we are creating the pages with hashEnabled false
+        List<Page> input = createPages(type, 5_000, 500, false);
+        RowPagesBuilder rowPagesBuilder = rowPagesBuilder(false, Ints.asList(0), type);
```

Comment:
This is redundant. Only the `hashChannel` is needed. Check the constructor of `SetBuilderOperatorFactory` and set the number would be good enough.